app/jobs/idd2wdl.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+import os
+import json
+import argparse
+import collections
+
+parser = argparse.ArgumentParser("idd2wdl", description="Convert an idseq-dag DAG to a WDL workflow")
+parser.add_argument("dag")
+parser.add_argument("--name")
+args = parser.parse_args()
+workflow_name = args.name or os.path.basename(args.dag).replace(".json", "")
+
+with open(args.dag) as fh:
+    dag = json.load(fh)
+
+print("version 1.0")
+
+input_names = collections.OrderedDict()
+targets_to_steps, step_class_count = dict(), collections.defaultdict(int)
+for step in dag["steps"]:
+    targets_to_steps[step["out"]] = step
+    step_class_count[step["class"]] += 1
+
+
+def file_path_to_name(f):
+    return f.replace(".", "_").replace("/", "_").replace("-", "_")
+
+
+def task_name(step):
+    if step_class_count[step["class"]] < 2:
+        name = step["class"]
+    else:
+        name = step["class"] + "_" + step["out"]
+    if name.startswith("PipelineStep"):
+        name = name[len("PipelineStep"):]
+    return name
+
+
+def target_filename(target, index):
+    if target == "fastqs":
+        return f"{target}_{index}"
+    else:
+        return f"{target}_{file_path_to_name(dag['targets'][target][index])}"
```

Comment:
So it appears that the Docker images we use have Debian 9.0 _stretch_, and the latest python3 version for that distro is `3.5.3`. Unfortunately f-strings require python3 v`3.6` or greater. We have to either rewrite this script without 3.6+ features or upgrade our Docker image to use the latest stable Debian 10.0 _buster_, which has python3 v`3.7`. Otherwise, as currently built trying to kickoff the sfn pipeline fails with

```
SfnPipelineDispatchService::Idd2WdlError (Command to convert dag to wdl failed ('idd2wdl.py'). Error:   File "app/jobs/idd2wdl.py", line 43)
    return f"{target}_{index}"
                             ^
SyntaxError: invalid syntax
```

As an aside, this is a good reason why running Docker locally is valuable for development, since we can reproduce the production environment and catch things like this. :)

Reply:
Good catch. Thanks! 
Point taken. Indeed, we are trading better performance for not fully recreating the prod environment. Nothing replaces the exact docker configuration that will be used in prod.

I still prefer to be on the side of performance for most vs. the occasional error. And there are other ways to provide "similar" guarantees. We can definitely improve our CI configuration, make sure we freeze versions when installing any packages and now that @kislyuk added a script to run it locally we can keep improving it to make sure it is synced with docker config.

Reply:
@lvreynoso unlike Docker on Linux, Docker on MacOS has performance issues that have their own negative impacts on development productivity. The reason is that Docker on MacOS uses full-blown virtualization, while Docker on Linux uses lightweight pseudo-virtualization that is much less resource intensive.

I agree that we should test our code in Docker to accurately represent production conditions. I think this should be done in a CI/CD pipeline that runs on GitHub Actions or in another CI environment, not on our laptops.

Reply:
I would add here that I believe Ubuntu LTS (not Debian) should be our preferred OS baseline. However, for this specific issue, I'm happy to rewrite the script to avoid the use of f-strings and be otherwise python3.5-compatible. I'll push a branch and link here when done.

Reply:
The Python 3.5 compatible copy of idd2wdl.py is now ready at https://github.com/chanzuckerberg/idseq/blob/akislyuk-idd2wdl-py35-compat/workflows/idd2wdl.py. It's definitely less readable and maintainable, but that shouldn't be a problem in the long term since our plan is to move away from generating DAG JSON and converting it to WDL, and toward using static WDL-defined workflows directly (which will remove the need for this script).

Reply:
I'm also a fan of Ubuntu! I did not know about the differences in Docker performance between Linux and MacOS, thanks for the information.

Reply:
@kislyuk what the factors in favor of Ubuntu LTS? Better support?

Reply:
Yes - Ubuntu LTS is better supported than Debian, has access to a wider variety of packages, and is used more widely in bioinformatics (making it more likely that we can standardize on it across all our environments).